sources/models/src/modeled_types/shared.rs

```diff
@@ -647,6 +647,127 @@ mod test_sysctl_key {
 
 // =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
+/// BootConfigKey represents a string that is a valid Kernel boot config keyword; Each key word must
+/// contain only alphabets, numbers, dash (-) or underscore (_).
+/// BootConfigKey stores the original string and makes it accessible through standard traits.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct BootConfigKeyword {
+    inner: String,
+}
+
+impl TryFrom<&str> for BootConfigKeyword {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, error::Error> {
+        let valid_keyword = input
+            .chars()
+            .all(|c| (c.is_ascii() && c.is_alphanumeric()) || c == '-' || c == '_')
+            && !input.is_empty();
+        ensure!(
+            valid_keyword,
+            error::InvalidBootconfigKeywordSnafu { input }
+        );
+        Ok(BootConfigKeyword {
+            inner: input.to_string(),
+        })
+    }
+}
+
+string_impls_for!(BootConfigKeyword, "BootConfigKeyword");
+
+#[cfg(test)]
+mod test_bootconfig_keyword {
+    use super::BootConfigKeyword;
+    use std::convert::TryFrom;
+
+    #[test]
+    fn valid_bootconfig_keyword() {
+        for ok in &[
+            "keyword",
+            "key_word",
+            "key-word",
+            "keyword1",
+            "keyword1-",
+            "keyword2_",
+        ] {
+            BootConfigKeyword::try_from(*ok).unwrap();
+        }
+    }
+
+    #[test]
+    fn invalid_bootconfig_keyword() {
+        for err in &[
+            "", "①", ".", "!", "@", "#", "$", "%", "^", "&", "*", "(", ")", "\"", "'", "\\", "|",
+            "~", "`",
+        ] {
+            BootConfigKeyword::try_from(*err).unwrap_err();
+        }
+    }
+}
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// BootConfigValue represents a string that is a valid Kernel boot config value; Each value only
+/// contains printable characters or spaces except for delimiters such as semicolon, newline, comma,
+/// hash, and closing brace. These delimiters are only usable if the value itself is quoted with
+/// single-quotes or double-quotes.
+/// BootConfigValue stores the original string and makes it accessible through standard traits.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct BootConfigValue {
+    inner: String,
+}
+
+impl TryFrom<&str> for BootConfigValue {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, error::Error> {
+        let quoted = (input.starts_with('"') && input.ends_with('"'))
+            || (input.starts_with('\'') && input.ends_with('\''));
+        let chars_that_require_quotes = ['\'', '"', '\n', ',', ';', '#', '}'];
+        let valid_value = input
```

Comment:
This is nice - I was wondering how this code was going to be to write - nice work!